system/security/LdapSecurity/ldapconnection.cpp

```diff
@@ -2247,6 +2257,17 @@ class CLdapClient : implements ILdapClient, public CInterface
 
     bool lookupAccount(MemoryBuffer& sidbuf, StringBuffer& account_name, ACT_TYPE& act_type)
     {
+        {
+            int sidLen = sidbuf.length();
+            ReadLockBlock readLock(m_unknownSIDCacheLock);
+            ForEachItemIn(idx, m_unknownSIDCache)
```

Comment:
How big is this list going to get? Is a linear search appropriate?

Reply:
@richardkchapman In my testing using the SDS Dali and real users, it never grew above 54 entries, and they were of varying lengths so I rarely had to do a few memcmp(). I will do some more profiling, but this simple

Reply:
@richardkchapman I ran a single user permissions query, and found this cache was populated with 54 entries, and a cache hit occurred over 17,000 times, eliminating 17,000 needless LDAP calls. Note the cache size did not increase when I queried multiple additional users.
@wangkx Please rereview based on your comments

Reply:
Actually, this eliminates 17,000*3= 51,000 unnecessary LDAP calls, since for each lookup it first searches the userBaseDN, then the GroupsBaseDN, and finally the SysUserBaseDN